### PR TITLE
People applied value

### PR DESCRIPTION
**Pull request by g4s8**
#653 - added people applied element access and join `@todo`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#881](https://github.com/zerocracy/farm/issues/881) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#881](https://github.com/zerocracy/farm/issues/881) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/921141656); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/com/zerocracy/pmo/People.java [hunk 1]
```diff
@@ -707,6 +709,70 @@ public boolean exists(final String pid) throws IOException {
...
+     * @return Applied time (UTC)
+     * @throws IOException If fails
+     */
+    public Date appliedTime(final String uid) throws IOException {
+        if (!this.exists(uid)) {
+            throw new IllegalArgumentException(
+                new Par("Person @%s doesn't exist").say(uid)
+            );
+        }
+        try (final Item item = this.item()) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/People.java]: @g4s8 we also need to check `!this.applied(uid)` condition here.

File: src/test/java/com/zerocracy/pmo/PeopleTest.java [hunk 2]
```diff
@@ -264,4 +266,23 @@ public void getSingleLink() throws Exception {
             Matchers.equalTo(href)
         );
     }
+
+    @Test
+    public void applyTest() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/PeopleTest.java]: @g4s8 let's name this test `canApply`.

Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/PeopleTest.java]: @g4s8 let's add a couple of test for negative cases as well:

* User applies but does not exist
* Get apply date of user that has not applied.

**Comment by g4s8** [comment]
@carlosmiranda fixed, please see 625d79e

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by g4s8** [comment]
@yegor256 it's `SignatureTest.rendersSignature` again, please restart merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by g4s8** [comment]
@yegor256 resolved conflicts, please restart merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by g4s8** [comment]
@yegor256 `SignatureTest.rendersSignature` failed, please restart merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by g4s8** [comment]
@carlosmiranda @yegor256 can't merge because of this: #943

**Comment by g4s8** [comment]
@yegor256 please restart merge, I merged `master` with #943 fix

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#881](https://github.com/zerocracy/farm/issues/881) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
